lib/grok-pure.rb

```diff
@@ -60,6 +60,9 @@ def initialize
   public
   def add_pattern(name, pattern)
     @logger.info("Adding pattern", name => pattern)
+    if @logger.debug? and pattern.end_with?("\s")
```

Comment:
I didn't expect this to work, but apparently \s in ruby strings means literally a space (hex 0x20).

I think we want it more broadly to say any whitespace -

`if pattern =~ /\s$/`

Which will match trailing tabs, spaces, etc.